### PR TITLE
feat: add Task API to sdk.entry

### PR DESCRIPTION
**Pull request by VaguelySerious**
# Purpose of PR

Adds methods that allow accessing/updating `Task` entities on an `Entry`

## PR Checklist

- [x] Tests are added/updated/not required
- [x] Tests are passing
- [x] Typescript typings are added/updated/not required

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -136,5 +136,6 @@
       ],
       "@semantic-release/github"
     ]
-  }
+  },
+  "dependencies": {}
```
Comment (andipaetzold) [inline package.json:140]: Just curious: Is this change required?

File: lib/types.ts [hunk 2]
```diff
@@ -131,7 +139,44 @@ export interface EntryFieldAPI {
...
+  sys: TaskSys
+}
+
+/** Allows accessing the Task API for the current entry. */
+export interface TaskAPI {
+  getTask(id: string): Promise<Task>
+  getTasks(): Promise<CollectionResponse<Task>>
+  createTask(data: TaskInputData): Promise<Task>
+  updateTask(task: Task): Promise<Task>
+  deleteTask(task: Task): Promise<void>
```
Comment (andipaetzold) [inline lib/types.ts:176]: The delete calls in `SpaceAPI` seem to return the actual object that was deleted.
Any reason for not doing that here?

File: lib/types.ts [hunk 3]
```diff
@@ -131,7 +139,44 @@ export interface EntryFieldAPI {
...
+}
+
+export interface EntryAPI extends TaskAPI {
```
Comment (andipaetzold) [inline lib/types.ts:179]: Not happy with extending `EntryAPI` here as we can only extend from one interface. This will become problematic when adding new apis to the entry in the future.

I'd rather add all functions of `TaskAPI` directly into `EntryAPI`

File: test/unit/entry.spec.ts [hunk 4]
```diff
@@ -65,6 +70,24 @@ describe('createEntry()', () => {
...
+          const args = ['foo', 42, {}]
+          describeChannelCallingMethod({
+            creator: createEntry,
+            methodName,
+            channelMethod: 'callEntryMethod',
+            args: args,
+            expectedCallArgs: [methodName, args],
+            makeCreatorArgs: () => {
+              return [{}, [{}], () => ({})]
+            },
```
Comment (andipaetzold) [inline test/unit/entry.spec.ts:85]: `makeCreatorArgs` is confusing to me.
How about

```javascript
creator: (channel) => createEntry(channel, {}, [{}], () => ({}))
```

and removing that again. It feels like it makes it unnecessarily complex

**Review by andipaetzold — CHANGES_REQUESTED** [review]
This should work. Still some comments.

Also, we need to add some integration tests before mergin.

File: lib/types.ts (hunk 3 above)
Comment (VaguelySerious) [reply lib/types.ts:179]: `interface A extends B, C, D {}` works?
And the reason for the split is that it made it easier to add the methods in `entry.ts`
```js
  const taskApi = {} as TaskAPI
  ...
  return {
    otherThings,
   ...taskApi
   }
```
But it's also easily possible to do this without splitting the interface, so if you still feel strongly about changing it let me know

Comment (andipaetzold) [reply lib/types.ts:179]: Oh - didn't know extending multiple interfaces

File: package.json (hunk 1 above)
Comment (VaguelySerious) [reply package.json:140]: No, but npm added this for me, so I assume it's good practice

File: lib/types.ts (hunk 2 above)
Comment (VaguelySerious) [reply lib/types.ts:176]: I was going off of the CMA docs. This seems to be a mistake either in the docs, or in the SpaceAPI then, good catch! I'll investigate

Comment (VaguelySerious) [reply lib/types.ts:176]: So this code is ok for now, but we'll need to change SpaceAPI later

File: test/unit/space.spec.ts [hunk 5]
```diff
@@ -57,6 +57,7 @@ describe('createSpace()', () => {
         channelMethod: 'callSpaceMethod',
         args: args,
         expectedCallArgs: [spaceMethod, args],
+        makeCreatorArgs: () => [],
```
Comment (Jwhiles) [inline test/unit/space.spec.ts:60]: Can this now be removed? Looks like you removed the associated code.

File: test/cypress/integration/sidebar-extension.spec.ts [hunk 6]
```diff
@@ -71,5 +77,30 @@ context('Sidebar extension', () => {
     })
   })
 
+  it('creates, gets, and deletes a content type through spaceAPI', () => {
```
Comment (VaguelySerious) [inline test/cypress/integration/sidebar-extension.spec.ts:80]: I didn't want to add _all_ the methods for _all_ the entities on space API, because we're not testing the CMA implementation, instead tried to get a full basic example in, so we at least know the integration is working on an SDK level.

**Review by Jwhiles — APPROVED** [review]

File: test/cypress/integration/sidebar-extension.spec.ts [hunk 7]
```diff
@@ -71,5 +77,27 @@ context('Sidebar extension', () => {
...
+    cy.getSdk(iframeSelector).then(async (sdk: SidebarExtensionSDK) => {
```
Comment (andipaetzold) [inline test/cypress/integration/sidebar-extension.spec.ts:81]: I personally don't like the mix of `.then` and `async`. Either one or the other.

**Review by andipaetzold — APPROVED** [review]

File: test/cypress/integration/sidebar-extension.spec.ts (hunk 7 above)
Comment (VaguelySerious) [reply test/cypress/integration/sidebar-extension.spec.ts:81]: It's not possible to `await` the `getSdk` call, sorry!
